### PR TITLE
added shell sort algorithm in go

### LINKED ISSUE #2156: Shell Sort
- [x] C
- [x] C++
- [x] C# - @Harikrishnan6336
- [x] CoffeeScript
- [ ] Dart
- [x] Go - @lakshyajit165
- [x] Java
- [x] JS
- [ ] Kotlin - @myselfanuj
- [x] Php - @saloniankita
- [x] Python
- [x] Readme
- [x] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by lakshyajit165**
Fixes #2156 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Shell sort algorithm in Go

#### Languages Used:
- Go

#### Files Added:
- Shell_Sort.go

Thanks!

### CODE REVIEW

**Comment by lakshyajit165** [comment]
@jainaman224, @raghav-dalmia @asha15. Please review my PR.

File: Shell_Sort/Shell_Sort.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+    "fmt"
+    "strconv"
+)
+
+func main() {
+  var n int
+  fmt.Scanf("%d", &n)
```
Comment (asha15) [inline Shell_Sort/Shell_Sort.go:10]: please correct the indentation

File: Shell_Sort/Shell_Sort.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  // input the size of the array
+
```
Comment (asha15) [inline Shell_Sort/Shell_Sort.go:13]: remove this line

File: Shell_Sort/Shell_Sort.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  var arr = make([]int, n)
+
+  // input the elements of the array
+
+  for i := 0; i < n; i++ {
+   
+        if _, err := fmt.Scan(&arr[i]); err != nil {
+            panic(err)
+        }
```
Comment (asha15) [inline Shell_Sort/Shell_Sort.go:22]: please remove the extra lines and correct the indentation

**Comment by lakshyajit165** [comment]
@asha15 and @jainaman224, I have made the required changes. Please review.

**Pull request merged**